polaris-service/src/main/java/org/apache/polaris/service/PolarisApplication.java

```diff
@@ -226,7 +230,17 @@ public void run(PolarisApplicationConfig configuration, Environment environment)
         new PolarisCallContextCatalogFactory(
             entityManagerFactory, metaStoreManagerFactory, taskExecutor, fileIOFactory);
 
-    PolarisAuthorizer authorizer = new PolarisAuthorizerImpl(configurationStore);
+    ConcurrentHashMap<RealmContext, EntityCache> realmEntityCache = new ConcurrentHashMap<>();
+    PolarisGrantManager.Factory factory =
+        new EntityCacheGrantManager.EntityCacheGrantManagerFactory(
```

Comment:
`PolarisGrantManager` objects returned from this factory are very different from what would be returned from `MetaStoreManagerFactory.getGrantManagerForRealm()`... What's the reason for having `getGrantManagerForRealm` in `MetaStoreManagerFactory`? I do not really see any call paths leading to the latter factory method.

Reply:
The `metastoreManagerFactory` passed to the constructor on the next line implements the `PolarisGrantManager.Factory`. The factory method is invoked by the `EntityCacheGrantManagerFactory` constructor here: https://github.com/apache/polaris/pull/465/files/3b10cb2021cecfc85dc93b2ed51ea3a4489439e4#diff-dec23d1ca2299b42bd2926a823cad0da6f0a8334ae5093346467f410280e77a2R69

Reply:
The `getGrantManagerForRealm` method implemented by this factory calls `delegateGrantManagerFactory.getGrantManagerForRealm(realm)`, which is invoking the `getGrantManagerForRealm` method on the `MetaStoreManagerFactory`. The `EntityCacheGrantManagerFactory` constructor takes in a `PolarisGrantManager.Factory` argument - here, we're passing in the `MetaStoreManagerFactory` as the delegate implementation.

Reply:
It looks like this code got refactored since my first comment (commit b9ded774360a785a2543dce26505e9039e0b2e42).